server/src/main/java/io/crate/expression/scalar/arithmetic/SubtractTimestampScalar.java

```diff
@@ -0,0 +1,85 @@
+ * additional information regarding copyright ownership.  Crate licenses
+ * this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.expression.scalar.arithmetic;
+
+import java.util.List;
+import java.util.function.BiFunction;
+
+import org.joda.time.Period;
+
+import io.crate.data.Input;
+import io.crate.expression.scalar.ScalarFunctionModule;
+import io.crate.metadata.NodeContext;
+import io.crate.metadata.Scalar;
+import io.crate.metadata.TransactionContext;
+import io.crate.metadata.functions.BoundSignature;
+import io.crate.metadata.functions.Signature;
+import io.crate.types.DataTypes;
+
+public class SubtractTimestampScalar extends Scalar<Period, Object> implements BiFunction<Long, Long, Period> {
+
+    public static void register(ScalarFunctionModule module) {
+        for (var timestampType : List.of(DataTypes.TIMESTAMP, DataTypes.TIMESTAMPZ)) {
+            module.register(
+                Signature.scalar(
+                    ArithmeticFunctions.Names.SUBTRACT,
+                    timestampType.getTypeSignature(),
+                    timestampType.getTypeSignature(),
+                    DataTypes.INTERVAL.getTypeSignature()
+                ),
+                SubtractTimestampScalar::new
+            );
+        }
+    }
+
+    private final Signature signature;
+    private final BoundSignature boundSignature;
+
+    public SubtractTimestampScalar(Signature declaredSignature,
+                                     BoundSignature boundSignature) {
+        this.signature = declaredSignature;
+        this.boundSignature = boundSignature;
+    }
+
+    @Override
+    public Signature signature() {
+        return signature;
+    }
+
+    @Override
+    public BoundSignature boundSignature() {
+        return boundSignature;
+    }
+
+    @Override
+    public Period evaluate(TransactionContext txnCtx, NodeContext nodeCtx, Input<Object>[] args) {
+        return apply((Long) args[1].value(), (Long) args[0].value());
+    }
+
+    @Override
+    public Period apply(Long start, Long end) {
+        if (end == null || start == null) {
+            return null;
+        }
+        return new Period(end - start);
+    }
```

Comment:
```suggestion
    public Period evaluate(TransactionContext txnCtx, NodeContext nodeCtx, Input<Object>[] args) {
        Long start = (Long) args[0].value();
        Long end = (Long) args[1].value();
        if (end == null || start == null) {
            return null;
        }
        return new Period(end - start);
    }
```

Looks like the `BiFunction` implementation is not used?

Reply:
ah yes, thx, that's why I kept apply there, sorry.